input/otlp/logs.go

```diff
@@ -239,3 +245,28 @@ func (c *Consumer) convertLogRecord(
 
 	return event
 }
+
+// https://www.elastic.co/guide/en/ecs/current/ecs-data_stream.html
+func sanitizeDataStream(k string) string {
+	k = strings.ToLower(k)
+	if strings.ContainsAny(k, DisallowedDataStreamRunes) {
+		k = strings.Map(replaceReservedRune, k)
+	}
+	// Cannot start with  _, +
+	// https://github.com/elastic/ecs/blob/main/rfcs/text/0009-data_stream-fields.md
+	if k[0] == '_' || k[0] == '+' {
+		k = k[1:]
+	}
+	if utf8.RuneCountInString(k) > MaxDataStreamRunes {
+		return string([]rune(k)[:MaxDataStreamRunes])
+	}
+	return k
+}
+
+func replaceReservedRune(r rune) rune {
+	switch r {
+	case '-', '\\', '/', '*', '?', '"', '<', '>', '|', ' ', ',', '#':
```

Comment:
Having a list of disallowed runes in this switch case, as well as in DisallowDataStreamRunes creates 2 sources of truth and will be harder to maintain down the road, especially when the 2 lines are separated by 200+ lines.